Content.Server/GameObjects/Components/Atmos/Piping/Pumps/VolumePumpComponent.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Content.Server.Atmos;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+
+namespace Content.Server.GameObjects.Components.Atmos.Piping.Pumps
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(BasePumpComponent))]
+    public class VolumePumpComponent : BasePumpComponent
+    {
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int VolumePumpRate
+        {
+            get => _volumePumpRate;
+            set => _volumePumpRate = Math.Clamp(value, 0, MaxVolumePumpRate);
+        }
+        private int _volumePumpRate;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int MaxVolumePumpRate
+        {
+            get => _maxVolumePumpRate;
+            set => Math.Max(value, 0);
+        }
+        private int _maxVolumePumpRate;
+
+        public override string Name => "VolumePump";
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _volumePumpRate, "startingVolumePumpRate", 0);
+            serializer.DataField(ref _maxVolumePumpRate, "maxVolumePumpRate", 100);
+        }
+
+        protected override void PumpGas(GasMixture inletGas, GasMixture outletGas)
+        {
+            var volumeRatio = Math.Clamp(VolumePumpRate / inletGas.Volume, 0, 1);
+
+            var transferedVolume = Math.Min(VolumePumpRate, inletGas.Volume);
+            var transferingGas = new GasMixture(transferedVolume);
+            transferingGas.Merge(inletGas);
+            transferingGas.Multiply(volumeRatio);
+
+            inletGas.Remove(transferingGas.TotalMoles);
+
+            outletGas.Merge(transferingGas);
```

Comment:
Why don't you use `inletGas.RemoveRatio` here instead? To get the ratio you would divide the transfer rate by the total volume of the inlet gas. That way you can avoid creating a new gas mixture for the sole purpose of calculating the moles to be removed from the inlet.